src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -35,5 +37,39 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::RequestCommissioning(int commissionerNumber)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY_CLIENT
+    const Mdns::DiscoveredNodeData * commissioner;
+    for (int i = 0; i < CHIP_DEVICE_CONFIG_MAX_DISCOVERED_NODES; i++)
+    {
+        commissioner = GetDiscoveredCommissioner(i);
+        if (commissioner != nullptr && --commissionerNumber == 0)
```

Comment:
`--commissionerNumber == 0` is not an obvious condition. What is being done here?

Reply:
Added documentation inline in the code.